File: Exercices/Séance 1 - C# vs Java-20231230/ActorsDB/src/be/ipl/domaine/Actor.java
package be.ipl.domaine;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.Calendar;
import java.util.Iterator;
import java.util.List;

public class Actor extends Person implements Serializable {

	private static final long serialVersionUID = 1L;
	private final int sizeInCentimeter;
	private List<Movie> movies;

	public int getSizeInCentimeter() {
		return sizeInCentimeter;
	}

	
	
	public Actor(String name, String firstname, Calendar birthDate, int sizeInCentimeter) {
		super(name, firstname, birthDate);
		this.sizeInCentimeter = sizeInCentimeter;
		movies = new ArrayList<Movie>();
	}


	@Override
	public String toString() {
		return "Actor [name = " + getName() + " firstname = " + getFirstname() + " sizeInCentimeter = " + sizeInCentimeter + " birthdate = " + getBirthDate() + "]";
	}
	
	/**
	 * 
	 * @return list of movies in which the actor has played
	 */
	public Iterator<Movie> Movies(){
		return movies.iterator();
	}
	
	/**
	 * add movie to the list of movies in which the actor has played
	 * @param movie
	 * @return false if the movie is null or already present
	 */
	public boolean addMovie(Movie movie){
		if ((movie == null) || movies.contains(movie))
			return false;
		
		if (!movie.containsActor(this))
			movie.addActor(this);

		movies.add(movie);

		return true;
	}

	public boolean containsMovie(Movie movie){
		return movies.contains(movie);
	}

	@Override
	public String getName(){
		return super.getName().toUpperCase();
	}
}

File: Exercices/Séance 1 - C# vs Java-20231230/Ex1/Actor.cs
namespace be.ipl.domaine;

[Serializable]
// implements or extend -> :
internal class Actor : Person
{

	private readonly int _sizeInCentimeter;
	private List<Movie> _movies;

	public int SizeInCentimeter() { return _sizeInCentimeter; }

	// super -> base
	public Actor(string name, string firstname, DateTime birthDate, int sizeInCentimeter) : base(name, firstname, birthDate)
	{
		this._sizeInCentimeter = sizeInCentimeter;
		this._movies = new List<Movie>();
	}


	public override string ToString()
	{
		return "Actor [name = " + Name + " firstname = " + Firstname + " sizeInCentimeter = " + SizeInCentimeter() + " birthdate = " + BirthDate + "]";
	}

	/**
	 * 
	 * @return list of movies in which the actor has played
	 */
	public IEnumerator<Movie> Movies()
	{
		return _movies.GetEnumerator();
	}

	/**
	 * add movie to the list of movies in which the actor has played
	 * @param movie
	 * @return false if the movie is null or already present
	 */
	public bool AddMovie(Movie movie)
	{
		if ((movie == null) || _movies.Contains(movie))
			return false;

		if (!movie.ContainsActor(this))
			movie.AddActor(this);

		_movies.Add(movie);

		return true;
	}

	public bool ContainsMovie(Movie movie)
	{
		return _movies.Contains(movie);
	}

	// to redefine a properties -> virtual in parent class + override in class child (here)
	public override string Name { get { return base.Name.ToUpper(); } }

}

